input.go

```diff
@@ -59,20 +115,24 @@ func InputsUTXORefsUniqueValidator() InputsValidatorFunc {
 	}
 }
 
-// InputsUTXORefIndexBoundsValidator returns a validator which checks that the UTXO ref index is within bounds.
-func InputsUTXORefIndexBoundsValidator() InputsValidatorFunc {
-	return func(index int, input *UTXOInput) error {
-		if input.TransactionOutputIndex < RefUTXOIndexMin || input.TransactionOutputIndex > RefUTXOIndexMax {
+// InputsSyntacticalIndicesWithinBounds returns an InputsSyntacticalValidationFunc which checks that the UTXO ref index is within bounds.
+func InputsSyntacticalIndicesWithinBounds() InputsSyntacticalValidationFunc {
+	return func(index int, input Input) error {
+		ref, is := input.(IndexedUTXOReferencer)
+		if !is {
+			return nil
+		}
+		if ref.Index() < RefUTXOIndexMin || ref.Index() > RefUTXOIndexMax {
 			return fmt.Errorf("%w: input %d", ErrRefUTXOIndexInvalid, index)
 		}
 		return nil
 	}
 }
 
-var utxoInputRefBoundsValidator = InputsUTXORefIndexBoundsValidator()
+var inputsPredicateIndicesWithinBounds = InputsSyntacticalIndicesWithinBounds()
 
-// ValidateInputs validates the inputs by running them against the given InputsValidatorFunc.
-func ValidateInputs(inputs serializer.Serializables, funcs ...InputsValidatorFunc) error {
+// ValidateInputs validates the inputs by running them against the given InputsSyntacticalValidationFunc(s).
+func ValidateInputs(inputs Inputs, funcs ...InputsSyntacticalValidationFunc) error {
```

Comment:
Maybe add the Syntactical to the function name to make it more clear?
Just a suggestion, I know its also part of the comment.

Reply:
276d8c15ffc6a9152f21386980dfc68f72ff5d0e